blocks/api/registration.js

```diff
@@ -113,13 +118,15 @@ export function registerBlockType( name, settings ) {
 	if ( ! settings.icon ) {
 		settings.icon = 'block-default';
 	}
-	const block = blocks[ name ] = {
+	settings = {
 		name,
 		attributes: get( window._wpBlocksAttributes, name ),
 		...settings,
 	};
 
-	return block;
+	settings = applyFilters( 'registerBlockType', settings, name );
```

Comment:
Should we revert the arguments to match the `registerBlockType` function signature? I know it's not convenient for the anchor filter because the name is useless.

Reply:
It's awkward, I agree, but the behavior of filtering is such that we need the second argument to be the original value we expect to be filtered; in this case, the settings. All additional arguments are considered "extra", and in this case the name could prove valuable and is not otherwise made available.

See: https://developer.wordpress.org/reference/functions/apply_filters/